### PR TITLE
Add testing helpers and benchmarks

### PR DESCRIPTION
**Pull request by josephlr**
When testing go-tpm-tools on real hardware, I noticed that some operations take much more time to complete than others. There is also a great difference in the amount of time needed between real hardware and virtualized hardware on Google cloud. This PR should make getting performance numbers much easier.

First, a new package called `internal` is added to contain testing helper code (so that we can use it with multiple packages in go-tpm-tools, but external users can't inadvertently bring it in). Then the testing code is cleaned up to use the internal package and benchmarks.

This allows very easy benchmark testing. For example (on Linux):
```bash
# Compile the benchmarking code
-> go test ./tpm2tools -c

# Run against the simulator
-> ../tpm2tools.test -test.bench=. -test.run=^$
goos: linux
goarch: amd64
pkg: github.com/google/go-tpm-tools/tpm2tools
BenchmarkEndorsementKeyRSA-8   	     200	  14541982 ns/op
BenchmarkStorageRootKeyRSA-8   	      50	  29673783 ns/op
PASS

# Run against real hardware
-> sudo ./tpm2tools.test -test.bench=. -test.run=^$ -tpm-path=/dev/tpm0
goos: linux
goarch: amd64
pkg: github.com/google/go-tpm-tools/tpm2tools
BenchmarkEndorsementKeyRSA-8   	       1	3318950780 ns/op
BenchmarkStorageRootKeyRSA-8   	       1	7106487932 ns/op
PASS
```

### CODE REVIEW

File: internal/test_linux.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	"flag"
+	"io"
+	"testing"
+
+	"github.com/google/go-tpm/tpm2"
+
+	"github.com/google/go-tpm-tools/simulator"
+)
+
+var tpmPath = flag.String("tpm-path", "", "Path to Linux TPM character device (i.e. /dev/tpm0 or /dev/tpmrm0). Empty value (default) will run tests against the simulator.")
```
Comment (stevenrutherford) [inline internal/test_linux.go]: Nit: underscores for flag names are standard for most google-ish binaries.

File: internal/test_linux.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+		rwc, err := tpm2.OpenTPM(*tpmPath)
+		if err != nil {
+			tb.Fatalf("Opening TPM at %s failed: %v", *tpmPath, err)
+		}
+		return rwc
+	}
+
+	simulator, err := simulator.Get()
+	if err != nil {
+		tb.Fatalf("Initializing simulator failed: %v", err)
```
Comment (stevenrutherford) [inline internal/test_linux.go]: "simulator initialization failed" or "initialization of simulator failed".

also, avoid caps at the start of errors (unless it's a proper noun).

**Comment by stevenrutherford** [comment]
Looks good. A few nits.

File: internal/test_linux.go (hunk 2 above)
Comment (josephlr) [reply internal/test_linux.go]: Fixed here and elsewhere. However per [go style guidelines](https://chromium.googlesource.com/chromiumos/platform/tast/+/HEAD/docs/writing_tests.md#formatting) errors never start with caps, but log statements (like those in `t.Fatal` or `b.Error`) always start with caps.

File: internal/test_linux.go (hunk 1 above)
Comment (josephlr) [reply internal/test_linux.go]: Done, here and elsewhere

File: tpm2tools/keys_test.go [hunk 3]
```diff
@@ -42,3 +42,33 @@ func TestCreateSigningKeysInHierarchies(t *testing.T) {
...
+	rwc := internal.GetTPM(b)
+	defer rwc.Close()
+	for n := 0; n < b.N; n++ {
+		b.StartTimer()
+		key, err := EndorsementKeyRSA(rwc)
+		b.StartTimer()
+		if err != nil {
+			b.Fatal(err)
+		}
+		key.Close()
```
[the remark this answers is not in the document]
Comment (josephlr) [reply tpm2tools/keys_test.go]: Done, here and elsewhere

**Pull request merged**